nodejs/eks/workerpool.ts

```diff
@@ -0,0 +1,364 @@
+ */
+export interface WorkerPoolOptions {
+    /**
+     * The VPC in which to create the worker pool.
+     */
+    vpcId: pulumi.Input<string>;
+
+    /**
+     * The IDs of the cluster subnets to filter for worker pool subnet ids.
+     *
+     * If the list of subnets includes both public and private subnets, the Kubernetes API
+     * server and the worker nodes will only be attached to the private subnets. See
+     * https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html for more details.
+     */
+    clusterSubnetIds: pulumi.Input<string[]>;
+
+    /**
+     * The IDs of the explicit node subnets to attach to the worker pool.
+     *
+     * This option overrides clusterSubnetIds option.
+     */
+    nodeSubnetIds?: pulumi.Input<pulumi.Input<string>[]>;
+
+    /**
+     * The security group associated with the EKS cluster.
+     */
+    clusterSecurityGroup: aws.ec2.SecurityGroup;
+
+    /**
+     * The target EKS cluster.
+     */
+    cluster: aws.eks.Cluster;
+
+    /**
+     * The instance type to use for the cluster's nodes. Defaults to "t2.medium".
+     */
+    instanceType?: pulumi.Input<aws.ec2.InstanceType>;
+
+    /**
+     * The instance profile to use for all nodes in this workder pool.
+     */
+    instanceProfile: pulumi.Input<aws.iam.InstanceProfile>;
+
+    /**
+     * The eks node access config map used by worker nodes
+     */
+    eksNodeAccess: k8s.core.v1.ConfigMap;
+
+    /**
+     * The security group to use for all nodes in this workder pool.
+     */
+    nodeSecurityGroup?: aws.ec2.SecurityGroup;
+
+    /**
+     * Public key material for SSH access to worker nodes. See allowed formats at:
+     * https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html
+     * If not provided, no SSH access is enabled on VMs.
+     */
+    nodePublicKey?: pulumi.Input<string>;
+
+    /**
+     * The size in GiB of a cluster node's root volume. Defaults to 20.
+     */
+    nodeRootVolumeSize?: pulumi.Input<number>;
+
+    /**
+     * Extra code to run on node startup. This code will run after the AWS EKS bootstrapping code and before the node
+     * signals its readiness to the managing CloudFormation stack. This code must be a typical user data script:
+     * critically it must begin with an interpreter directive (i.e. a `#!`).
+     */
+    nodeUserData?: pulumi.Input<string>;
+
+    /**
+     * The number of worker nodes that should be running in the cluster. Defaults to 2.
+     */
+    desiredCapacity?: pulumi.Input<number>;
+
+    /**
+     * The minimum number of worker nodes running in the cluster. Defaults to 1.
+     */
+    minSize?: pulumi.Input<number>;
+
+    /**
+     * The maximum number of worker nodes running in the cluster. Defaults to 2.
+     */
+    maxSize?: pulumi.Input<number>;
+
+    /**
+     * The Amazon VPC CNI plugin for this worker pool's cluster.
+     */
+    vpcCni: VpcCni;
```

Comment:
I'm a bit unsure about passing `vpcCni` and `eksNodeAccess` as arguments; they're really something of an implementation detail. I'm not sure I have a better option, though.

Reply:
how about passing core or cluster as an argument? this is assuming you always have to associate a worker pool with a specific cluster.

Reply:
> how about passing core or cluster as an argument

I thought about that briefly. I like the idea of allowing these as arguments, but I'd love to leave this type open-ended enough that it could work with with a cluster created manually (i.e. a cluster provisioned with explicit AWS resources or managed outside of Pulumi). We could perhaps use a union type of `aws.eks.Cluster | Core | Cluster` for the `cluster` property and do different things based on that.

Reply:
sounds good 👍